internal/gatewayapi/translator.go

```diff
@@ -0,0 +1,515 @@
+package gatewayapi
+
+import (
+	"fmt"
+
+	"github.com/envoyproxy/gateway/pkg/ir"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+const (
+	KindGateway   = "Gateway"
+	KindHTTPRoute = "HTTPRoute"
+	KindService   = "Service"
+)
+
+// Cache is an interface that the Translator requires
+// for getting Gateway API and related resources.
+type Cache interface {
+	ListGateways() []*v1beta1.Gateway
+	ListHTTPRoutes() []*v1beta1.HTTPRoute
+	GetNamespace(name string) *v1.Namespace
+	GetService(namespace, name string) *v1.Service
+}
+
+// Translator translates Gateway API resources to the IR,
+// and computes status for Gateway API resources.
+type Translator struct {
+	gatewayClassName v1beta1.ObjectName
+}
+
+type TranslateResult struct {
+	Gateways   []*v1beta1.Gateway
```

Comment:
`Gateways` and `HTTPRoutes` is added to the o/p to highlight the successful inputs the got translated ?

Reply:
They're there because their status have been updated, which then needs to be written back to the API server. The details of exactly how that works can obviously change per #147 but for now just including them in the output struct.

Reply:
cool yah can be refactored as part of #147